commands/command_smudge.go

```diff
@@ -63,6 +63,15 @@ func delayedSmudge(s *git.FilterProcessScanner, to io.Writer, from io.Reader, q
 			q.Add(filename, path, ptr.Oid, ptr.Size)
 			return 0, true, ptr, nil
 		}
+
+		// Write 'statusFromErr(nil)', even though 'perr != nil', since
+		// we are about to write non-delayed smudged contents to "to".
```

Comment:
The fix seems to work! Thank you 👍! However, I still try to decipher the comment 😄. `perr` should be `nil` here, no? TBH I don't really understand the same comment a few lines above either. Can you give me a hint?